src/Components/CalendarDesign.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useContext } from 'react';
+import 'antd/dist/antd.css';
+import './style.css';
+import { Button, Calendar, Input } from 'antd';
+import { Modal } from 'antd';
+import DayCell from './DayCell';
+const CalendarDesign = (props) => {
+  const [title, setTitle] = useState('');
```

Comment:
A. The modal should be in a separate component. 
B. When the modal submits the data, it should handle that internally. Meaning:
The modal should have `onSubmit` property that is `(values: { title, ...anyMoreData }) => void`, so you don't have to store this state in the calendar when it really doesn't have anything to do with the calendar.